src/devices/Bmp180/Bmp180.cs

```diff
@@ -93,9 +94,10 @@ public Pressure ReadPressure()
         /// <returns>
         ///  Height in meters from the sensor
         /// </returns>
+        [ObsoleteAttribute("ReadAltitude is obsolete. Please use Iot.Device.Common.WeatherHelper instead.")]

```

Comment:
I wouldn't mark these guys as obsolete. It is fine if internally we call WeatherHelper ourselves, but it is still way easier to call `bmp180.ReadAltitud(SeaLevelPressure)` than calling `WeatherHelper.CalculateAltitude(ReadPressure(), seaLevelPressure, ReadTemperature());`

The same goes for all of the rest of the Obsolete attributes you added. The implementation change is great, but obsoleting all these members I don't agree with.

Reply:
The idea is that `ReadAltitude` will force reading the temperature and pressure from your sensors, the values of which you may already have available at hand. That is, if you want to display the temperature, pressure and altitude, using the WeatherHelper IS the way to go to avoid polling the sensors twice. The only time using ReadAltitude directly makes sense is when you only want to get the altitude and don't care for the actual readings. This is why I marked them as obsolete. Please let me know what you think now that I've explained the why.

Reply:
I get that, and I agree that the most optimum code in most cases would be to get the Altitude through the WeatherHelper, but my point is that performance and CPU is not the ONLY goal of our APIs. One of the other goals (and just as important as perf) is the ease of use, which is why I think that we should provide that option as well. We have a lot of examples in the framework as well where there are two ways of doing the exact same thing, One is very easy to use, and the other is harder but more performant, and we care about both. One example is writing text to a file, you could do it through File.WriteAllText(path, text) which is super easy to use but not super performant, or you could also do it through opening a stream and managing the writing and closing of the stream yourself. That would be more performant, but that doesn't mean that we want to remove or obsolete the WriteAllText API.

Reply:
OK I appreciate this (though the example you brought is unfair since this is essentially a 1:1 line of code substitution) and understand your fair comments. In actuality I hoped the tag would serve a second purpose which is to make developers already using it aware that there's now a more efficient alternative. I wish C# had a [Suggestion] attribute; it'd make more sense than an obsolete attribute.

Reply:
Right, I agree suggestion would be great, the problem with obsolete is that many people compile with `TreatWarningsAsErrors` so this would mean that they would have to use #pragma disable on their code in order to be able to call this method, and they might not really care about performance, so that is the part I don't like about obsolete.

Reply:
Done (pending commit)